_rules/menuitem-non-empty-name-m6b1q3.md

```diff
@@ -0,0 +1,153 @@
+---
+id: m6b1q3
+name: Menuitem has non-empty accessible name
+rule_type: atomic
+description: |
+  This rule checks that each element with a `menuitem` role has a non-empty accessible name.
+accessibility_requirements:
+  wcag20:4.1.2: # Name, Role, Value (A)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  - Accessibility Tree
+  - DOM Tree
+  - CSS Styling
+acknowledgments:
+  authors:
+    - Wilco Fiers
+---
+
+## Applicability
+
+This rule applies to HTML elements with a `menuitem` [semantic role][] that are [included in the accessibility tree][].
```

Comment:
This looks extremely focused. Would it make sense to extend to any `widget` role? (or is there some that don't require a name?)

This is definitely a working rule but it feels it's checking something extremely narrow and could easily be extended.

Reply:
I do think all widgets require an accessible name, but some apply to different SCs. This one can't even be merged with its parent role "command", since that includes links, which are also applicable to 2.4.4.

Reply:
Hmm, I guess we're back to the discussion at #1049 where we decided it was OK to have a rule with wider applicability and lesser mapping, plus an extra rule with narrow applicability for those special cases, even if it ends up with the same element failing 2 rules for the same reason.

See also https://www.w3.org/2020/02/13-act-r-minutes.html#x03 and https://github.com/act-rules/act-rules.github.io/issues/1049#issuecomment-585818622

Not a show blocker for the current PR, however.